### ISSUE TITLE
NVDA ignores aria-label on a parent element if a descendant element has aria-hidden="true"

### ISSUE THREAD

**Issue by cary-rowen** [nvaccess/nvda#19040]
### Brief summary

When an HTML element is given an `aria-label`, but a descendant of that element is marked with `aria-hidden="true"`, NVDA does not announce the `aria-label` of the parent element. Instead, it announces the content of the visible descendant nodes.

### Steps to reproduce

1.  Save the following minimal HTML code as a file.
2.  Open the file in a browser.
3.  Launch NVDA.
4.  Using browse mode, navigate to the table row with the down arrow key.

**Minimal Code Example:**

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>ARIA Bug Reproduction</title>
</head>
<body>

    <table>
        <tr aria-label="This is the full, correct label.">
            
            <td aria-hidden="true">This text is hidden from screen readers.</td>

            <td>123</td>

        </tr>
    </table>

</body>
</html>
```

### Actual behavior

NVDA announces: "123".

### Expected behavior

NVDA should announce the content of the `aria-label` on the `<tr>` element: "This is the full, correct label.".

### NVDA type

installed copy, portable copy, installer/launcher copy

### NVDA version

2025.3

### Have you tried any other versions of NVDA? If so, please report their behaviors.

No

### Windows version

Windows 10 22H2 (AMD64) build 19045.6396

### Name and version of other software in use when reproducing the issue

Chrome141.0.7390.55, Firefox143.0.3

### Other information about your system

Narrator works as expected

### Does the issue still occur after restarting your computer?

I have restarted my computer and the issue still occurs

### If NVDA add-ons are disabled, is your problem still occurring?

I have restarted NVDA with add-ons disabled and the issue still occurs

### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

I have run the COM Registration Fixing Tool and the issue still occurs

**Comment by cary-rowen** [comment]
I encountered this issue on the Microsoft forms sharing analysis results page.

**Comment by cary-rowen** [comment]
Fill out [this form](https://forms.microsoft.com/r/005zjYN58K) and view the corresponding [summary page](https://forms.microsoft.com/Pages/AnalysisPage.aspx?AnalyzerToken=MEzAtJTmeKiePyUmbeu1dnD6LHXf702M&id=DQSIkWdsW0yxEjajBLZtrQAAAAAAAAAAAANAAdNjlxxUMjNOWk5FUUY5WjVKN1NUUkNGTDNFWlVIWC4u), which can reproduce the issue.

**Comment by tareh7z** [comment]
Pretty sure this is a bug. Aria-hidden shouldn’t kill the parent’s aria-label.

**Comment by LeonarddeR** [comment]
I think what @jcsteh said in https://github.com/nvaccess/nvda/issues/19391#issuecomment-3662659849 might also apply here. I generally think setting an aria-label on a tr is a bad habit anyway.

**Comment by cary-rowen** [comment]
> I generally think setting an aria-label on a tr is a bad habit anyway.

I agree. It's clear that products like Microsoft are also rife with such quirks, and this is likely to become even more prevalent with the rise of LLM.

**Comment by jcsteh** [comment]
I think aria-hidden is a red herring here. Even if there weren't aria-hidden descendant content, you should see the same behaviour. The key point is that aria-label does not override the content of a table row in browse mode. If it did, aside from anything else, you would lose the ability to navigate the table with table navigation commands, since there would no longer be any cells to navigate. There are also valid cases where the label of a row should include a summary of the information in the cells (e.g. for focus mode), but the cells contain content that you want in browse mode. Gmail message lists are a great example of this and we do not want to break those.

**Comment by seanbudd** [comment]
Was this fixed by #19409?

**Comment by cary-rowen** [comment]
I tested with Firefox and Chrome, and #19409 could not fix this issue.
